python/lsst/analysis/ap/apdb.py

```diff
@@ -211,41 +211,49 @@ def load_sources(self, exclude_flagged=False, limit=100000):
         self._fill_from_ccdVisitId(result)
         return result
 
-    def load_object(self, id):
-        """Load one diaObject.
+    def load_object(self, id, latest=True):
+        """Load one DiaObject.
 
         Parameters
         ----------
         id : `int`
             DiaObjectId to load data for.
+        latest : `bool`
+            Only load the DiaObject where validityEnd is None.
+            This is the most-recently updated DiaObject.
 
         Returns
         -------
         data : `pandas.Series`
             The requested object.
         """
         table = self._tables["DiaObject"]
-        query = table.select().where(table.columns["validityStart"].isnot(None))
+        if latest:
+            query = table.select().where(table.columns["validityEnd"] == None)  # noqa: E711
```

Comment:
I think you need both of these selects: we want a valid start time (we can withdraw sources by setting `validityStart=None`), and, depending on `latest`, either a valid end or any end.

Reply:
I do not believe it is possible for `validityStart` to be invalid. There is always a start time. There is not always an end time. If we want to do validation on validityStart ranges, that belongs in a unit test (runs away), not in the query call. For this function to return the expected list of "latest" valid DiaObjects, this is the correct query.

Reply:
Ah, `validityStart` is non-nullable (just checked sdm_schemas), that's true. I don't know whether we have a system for marking a DiaObject "withdrawn", but that's related to the new `time_processed` and `time_withdrawn` fields for DiaSources. I withdraw my objection about this change.

Pinging @ebellm (original author of this line) as a reminder of this.

Reply:
Withdrawing a DIAObject just means giving it a non-null `validityEnd`.